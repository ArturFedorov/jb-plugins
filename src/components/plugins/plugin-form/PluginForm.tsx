import React, { FunctionComponent } from 'react';
import styles from './plugin-form.module.scss';
import { Input } from '../../common/input/Input';
import { Button } from '../../common/button/Button';
import { ButtonType } from '../../common/button/ButtonType';

export const PluginForm: FunctionComponent = () => {
  const onChange = (val: string) => console.log(val);
  return (
    <form className={styles.pluginForm}>
      <div>
        <label>Name</label>
        <Input withBorder={true} onChange={onChange} />
      </div>
      <div>
        <label>Author name</label>
        <Input withBorder={true} onChange={onChange} />
      </div>
      <div>
        <label>Icon URL</label>
        <Input withBorder={true} onChange={onChange} />
      </div>
      <div>
        <Button type={ButtonType.ACTION}>Add plugin</Button>
        <Button type={ButtonType.DEFAULT}>Cancel</Button>
      </div>
    </form>
  );
};
