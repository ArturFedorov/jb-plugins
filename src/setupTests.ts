// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import thunk from 'redux-thunk';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import createMockStore from 'redux-mock-store';

export const mockStore = createMockStore([thunk]);
configure({ adapter: new Adapter() });
