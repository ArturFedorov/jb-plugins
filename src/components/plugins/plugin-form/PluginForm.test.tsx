import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import PluginForm from './PluginForm';
import { mockStore } from '../../../setupTests';

describe('PluginForm', () => {
  const component = mount(
    <Provider store={mockStore()}>
      <PluginForm />
    </Provider>
  );

  it('should match a snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
