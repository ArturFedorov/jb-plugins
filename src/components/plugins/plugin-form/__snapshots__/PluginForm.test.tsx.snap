// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginForm should match a snapshot 1`] = `ReactWrapper {}`;
